src/lang/buffer.cc

```diff
@@ -310,7 +310,7 @@ Buffer Buffer::MakeSlice(Array<Expr> begins, Array<Expr> extents) const {
     // check if stride is needed.
     for (size_t i = 0; i < extents.size(); ++i) {
       if (!can_relax) {
-        if (!is_zero(begins[i]) ||
+        if (!is_zero(ir::Simplify(begins[i])) ||
```

Comment:
Would it be better to simplify `begins` and `extents` right after line 303? So probably line 322 could benefit from the simplified expressions?

Reply:
Good point